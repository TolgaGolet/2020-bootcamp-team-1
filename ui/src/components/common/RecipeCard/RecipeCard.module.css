.Card {
  margin: 6px;
  width: 100%;
  text-align: left;
  display: inline-block;
  max-width: 300px;
}

.Card .CardImage {
  height: 140px;
  object-fit: cover;
}

.Card > div {
  padding: 8px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

@media (min-width: 800px) {
  .Card {
    margin: 10px;
  }

  .Card .CardImage {
    height: 200px;
  }

  .Card > div {
    padding: 12px;
  }
}
