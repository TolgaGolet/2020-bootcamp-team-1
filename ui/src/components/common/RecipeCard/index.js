import React from "react";
import styles from "../Common.module.css";
import TimerSvg from "../../svgs/TimerSvg";
import Card from "react-bootstrap/Card";
import Ellipsis from "react-ellipsis-pjs";

const RecipeCard = ({ item: { title, prepTime, description, image } }) => {
  return (
    <Card className={styles.Card}>
      <Card.Img variant="top" src={image} className={styles.CardImage} />
      <Card.Body>
        <Ellipsis
          text={title}
          lines={2}
          style={{
            fontSize: "1.3rem",
            fontWeight: "bold",
            marginBottom: "0.75rem"
          }}
        />
        <p className="mb-2 text-muted card-subtitle">
          <TimerSvg />
          {prepTime}
        </p>
        <Ellipsis text={description} lines={3} style={{ marginBottom: 0 }} />
      </Card.Body>
    </Card>
  );
};

export default RecipeCard;
