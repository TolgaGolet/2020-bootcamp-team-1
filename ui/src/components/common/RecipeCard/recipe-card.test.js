import React from "react";
import { shallow } from "enzyme";
import RecipeCard from "./";

const CardBodyComponent = "CardBody";
const CardComponent = "Card";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useHistory: () => ({
    push: jest.fn()
  })
}));

const setup = (props = {}) => {
  const component = shallow(<RecipeCard {...props} />);
  return component;
};

describe("Fab Component", () => {
  it("Renders correctly", () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it("Renders without props", () => {
    const wrapper = setup();
    const component = wrapper.find(CardComponent);
    expect(component.length).toBe(1);
  });

  it("Renders recipe preparation title correctly", () => {
    const wrapper = setup({ item: { prepTime: "TEST" } });
    const titleText = wrapper
      .find(CardBodyComponent)
      .find("p")
      .text();
    expect(titleText).toContain("TEST");
  });
});
