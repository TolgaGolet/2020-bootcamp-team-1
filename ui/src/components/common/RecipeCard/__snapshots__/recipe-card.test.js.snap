// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fab Component Renders correctly 1`] = `ShallowWrapper {}`;
